docs/network-troubleshooting.adoc

```diff
@@ -0,0 +1,135 @@
+
+```
+readiness signaling protocol failed: [waiting for readiness timed out after: [2m0s]]"
+```
+
+== Troubleshooting checklist
+
+The cause of aforementioned problems may lie in either member of the keep.
+To make sure your client performs well, please pass through the following
+checklist.
+
+=== Check your client health
+Check your client is up and doesn't experience frequent restarts. If your clients
+goes down, it breaks the ongoing protocol execution for other keep members.
+This is especially relevant for cloud workloads where things like VMs preemption
+or Kubernetes pod relocation may take place. It's strongly recommended to enable
+<<./run-keep-ecdsa.adoc#9-metrics, metrics>>
+for your client and use a monitoring system which allows catching all issues quickly
+(see https://github.com/boar-network/keep-monitoring[Boar Network monitoring system]).
+Using monitoring tools specific for your cloud provider may also be a good choice.
+
+=== Check internet connectivity
+Check your client has a working and stable internet connection. Temporary
+outages may make your client invisible for other keep members. Other members
+won't be able to connect your client and perform protocol executions. Metrics
+can help monitor network connectivity as they return the number of connected
+peers. A significant fall of connected peers number may indicate network
+issues.
+
+=== Check Ethereum connectivity
+Check if your node has a working connection with their ETH node. Broken
+ETH connection means that your client becomes blind for all chain events
+taking place for keeps it is a member of. For example, it may not notice a
+starting key generation or signing process. ETH connectivity can be
+watched using metrics and a monitoring system
+
+=== Check your client fulfills the minimum stake rule
+Check your client has the minimum amount of KEEP token staked. If the staked
+amount drops below the minimum threshold, your client won't be able to
+connect other clients and all existing connections will be dropped.
+
+=== Make sure your client is publicly reachable
+Check if your client uses a publicly available IP or DNS and announces it
+in the network. In case a private address is used, your node won't be able
+to receive incoming connections though it may be able to initiate them on their
+own. Public availability can be checked using the `nmap` tool. Regarding
+announcing, you need to use the
+<<./run-keep-ecdsa.adoc#322-parameters, `LibP2P.AnnounceAddresses`>>
+config property. Last but not least, your firewall rules should allow the
+ingress traffic to reach your client.
+
+=== Check your client is able to reach the public internet
+Your client should be able to reach other clients over the public internet.
+Make sure your network routing, NAT and firewall setup allows your client
+to do so. If this requirement is not fulfilled, your client won't be able
+to initiate outgoing connections though it may be able to receive incoming
+ones.
```

Comment:
For this and public reachability, are these things that would be reflected in the metrics? That is, would it be reflected in the number of connected peers? Or is that only true if the node is both unreachable *and* unable to reach public internet?

Reply:
As you said, this is true only if the node is both unreachable and unable to reach the public internet. Both problems occurring separately may not be observable in metrics. If it's unreachable but can reach the public internet, for example, through a NAT gateway, it can still initiate connections on its own so connected peers will be non-zero. Conversely, when it is reachable but can't reach the public internet, it can still receive inbound connections.